quipucords/api/aggregate_report/model.py

```diff
@@ -37,62 +39,98 @@
     """Results of the aggregate report."""
 
+    report = models.OneToOneField(
+        "Report", models.CASCADE, related_name="aggregate_report", null=True
+    )
+
     # Note the lack of special reporting needs for Satellite or RHACS here.
     # At the time of this writing, Justin says this is correct and we have none.
 
-    # TODO Convert this to a full Django model that we can store and look up.
-
-    ansible_hosts_all: int = 0
-    ansible_hosts_in_database: int = 0
-    ansible_hosts_in_jobs: int = 0
-    inspect_result_status_failed: int = 0
-    inspect_result_status_success: int = 0
-    inspect_result_status_unknown: int = 0
-    inspect_result_status_unreachable: int = 0
-    instances_hypervisor: int = 0
-    instances_not_redhat: int = 0
-    instances_physical: int = 0
-    instances_unknown: int = 0
-    instances_virtual: int = 0
-    jboss_eap_cores_physical: int = 0
-    jboss_eap_cores_virtual: int = 0
-    jboss_eap_instances: int = 0
-    jboss_ws_cores_physical: int = 0
-    jboss_ws_cores_virtual: int = 0
-    jboss_ws_instances: int = 0
-    missing_cpu_core_count: int = 0
-    missing_cpu_socket_count: int = 0
-    missing_name: int = 0
-    missing_pem_files: int = 0
-    missing_system_creation_date: int = 0
-    missing_system_purpose: int = 0
-    openshift_cores: int = 0
-    openshift_operators_by_name: dict = field(default_factory=dict)
-    openshift_operators_by_kind: dict = field(default_factory=dict)
-    os_by_name_and_version: dict = field(default_factory=dict)
-    socket_pairs: int = 0
-    system_creation_date_average: date | None = None
-    vmware_hosts: int = 0
-    vmware_vm_to_host_ratio: float = 0
-    vmware_vms: int = 0
+    ansible_hosts_all = models.PositiveIntegerField(default=0, blank=True, null=True)
+    ansible_hosts_in_database = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    ansible_hosts_in_jobs = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    inspect_result_status_failed = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    inspect_result_status_success = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    inspect_result_status_unknown = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    inspect_result_status_unreachable = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    instances_hypervisor = models.PositiveIntegerField(default=0, blank=True, null=True)
+    instances_not_redhat = models.PositiveIntegerField(default=0, blank=True, null=True)
+    instances_physical = models.PositiveIntegerField(default=0, blank=True, null=True)
+    instances_unknown = models.PositiveIntegerField(default=0, blank=True, null=True)
+    instances_virtual = models.PositiveIntegerField(default=0, blank=True, null=True)
+    jboss_eap_cores_physical = models.PositiveIntegerField(
```

Comment:
I wonder if that should be int, and not float?

That value is derived from `cpu_core_count`, and `cpu_core_count` was [explicitly changed from int to float 7 years ago](https://github.com/quipucords/quipucords/commit/989dccd7677be6309ad1ecf9c59be5da368af2c0). We have a thread about it on Slack (September 19, 2024). Nobody in current team ever saw that field being anything else than integer, but we also can't rule out some random cloud or ancient RHEL patch revision returning float. Personally, I would err on the side of past team knowing something that we don't.

Same comment applies to `jboss_eap_cores_virtual`, `jboss_ws_cores_physical` and `jboss_ws_cores_virtual`.